### PR TITLE
feat(logs): Add error fingerprint attribute

### PR DESCRIPTION
**Pull request by fuzzybinary**
### What and why?

Allow users to provide custom error fingerprint attribute.

In Logs the attribute `Logs.Attributes.errorFingerprint` is added to any log call and added to the `LogEvent.Error.fingerprint` member. 

In RUM, the attribute `RUM.Attributes.errorFingterprint` is added to  `addError` calls and it is transferred to the `RUMError.finterprint` member

refs: RUM-2906

### How?

A special attribute (defined at `Logs.Attributes.errorFingerprint`) is read and moved during LogEvent creation

### Review checklist
- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)
- [ ] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [x] Run unit tests for Core, RUM, Trace, Logs, CR and WVT
- [ ] Run unit tests for Session Replay
- [x] Run integration tests
- [x] Run smoke tests
- [ ] Run tests for `tools/`

### CODE REVIEW

File: DatadogLogs/Sources/RemoteLogger.swift [hunk 1]
```diff
@@ -110,6 +110,7 @@ internal final class RemoteLogger: LoggerProtocol {
         let tags = self.tags
         var logAttributes = attributes
         let isCrash = logAttributes?.removeValue(forKey: CrossPlatformAttributes.errorLogIsCrash) as? Bool ?? false
+        let errorFingerprint = logAttributes?.removeValue(forKey: Logs.Attributes.errorFingerprint) as? String
```
Comment (ncreated) [inline DatadogLogs/Sources/RemoteLogger.swift:113]: question/ Shouldn't we check if `level` is `>= error` 🤔, so this is not applied to lower severities?

File: DatadogRUM/Sources/RUM.swift [hunk 2]
```diff
@@ -53,3 +53,11 @@ public enum RUM {
         rum.monitor.notifySDKInit()
     }
 }
+
+extension RUM {
+    /// Attributes that can be added to RUM calls that have special properies in Datadog.
+    public struct Attributes {
+        /// Add a custom fingerprint to the error in this log. Requires that the log be supplied with an Error.
```
Comment (ncreated) [inline DatadogRUM/Sources/RUM.swift:60]: blocking/ similar to logs; also it calls "log" not RUM event. Ultimately:
```suggestion
        /// Add a custom fingerprint to the RUM error.
        /// The value of this attribute must be a `String`.
```

File: IntegrationTests/Runner/Scenarios/RUM/ManualInstrumentation/SendRUMFixture3ViewController.swift [hunk 3]
```diff
@@ -5,12 +5,22 @@
...
         super.viewDidAppear(animated)
 
         rumMonitor.startView(key: "fixture3-vc", name: "SendRUMFixture3View")
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.05) {
+            rumMonitor.addError(
+                message: "Simulated view error with fingerprint",
+                source: .source, attributes: [
+                    RUM.Attributes.errorFingerprint: "fake-fingerprint"
+                ]
```
Comment (ncreated) [inline IntegrationTests/Runner/Scenarios/RUM/ManualInstrumentation/SendRUMFixture3ViewController.swift:21]: nit
```suggestion
                source: .source, 
                attributes: [
                    RUM.Attributes.errorFingerprint: "fake-fingerprint"
                ]
```

**Review by ncreated — CHANGES_REQUESTED** [review]
🎯 Well done 👍. There are some blocking points in public API comments.

Except in-line comments, what's our take on allowing fingerprint in [RUM resource errors](https://github.com/DataDog/dd-sdk-ios/blob/54fae19f7b45787301ae8d0b99d701fc0f69e9ba/DatadogRUM/Sources/RUMMonitor/Scopes/RUMResourceScope.swift#L278)?

File: DatadogLogs/Sources/RemoteLogger.swift (hunk 1 above)
Comment (fuzzybinary) [reply DatadogLogs/Sources/RemoteLogger.swift:113]: No I don't think so, as you can pass errors to any level.  I could check that an error was passed in though?

File: DatadogLogs/Sources/Feature/MessageReceivers.swift [hunk 4]
```diff
@@ -68,6 +68,7 @@ internal struct LogMessageReceiver: FeatureMessageReceiver {
                     level: log.level,
                     message: log.message,
                     error: log.error,
+                    errorFingerprint: nil,
```
[the remark this answers is not in the document]
Comment (fuzzybinary) [reply DatadogLogs/Sources/Feature/MessageReceivers.swift:71]: As discussed at standup, it sounds like we won't support fingerprinting through traces at the moment.

File: DatadogLogs/Sources/RemoteLogger.swift (hunk 1 above)
Comment (ncreated) [reply DatadogLogs/Sources/RemoteLogger.swift:113]: Ah, indeed 👍. All good 👌

**Review by ncreated — APPROVED** [review]
Looks good 👍

PS. I removed `IPHONEOS_DEPLOYMENT_TARGET` overrides in https://github.com/DataDog/dd-sdk-ios/pull/1722/commits/45229f76b11c2e465e71ddbd31f919f7a7f61c10 as they went again into the change 😉

**Pull request merged**